tests/contrib/django/test_middleware.py

```diff
@@ -63,6 +63,21 @@ def test_middleware_trace_function_based_view(self):
         eq_(span.get_tag('http.url'), '/fn-view/')
         eq_(span.resource, 'tests.contrib.django.app.views.function_view')
 
+    def test_middleware_trace_error_500(self):
+        # ensures that all error 500 are well traced
+        url = reverse('error-500')
+        response = self.client.get(url)
+        eq_(response.status_code, 500)
+
+        # check for spans
+        spans = self.tracer.writer.pop()
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.get_tag('http.status_code'), '500')
+        eq_(span.get_tag('http.url'), '/error-500/')
+        eq_(span.resource, 'tests.contrib.django.app.views.error_500')
+        assert "ZeroDivisionError: integer division or modulo by zero" in span.get_tag('error.stack')
```

Comment:
while this solution is quite good and handles exceptions with a two-phases middleware (like the built-in Django cache middleware), it doesn't cover errors raised in other middlewares. Suppose you have this use case:
```python
class MiddlewareBlast(object):
    def process_request(self, request):
        1 / 0  # i.e. a critical failure while asking data to another service

# in your settings
MIDDLEWARE_CLASSES = [
    'ddtrace.contrib.django.TraceMiddleware',
    'tests.contrib.django.app.middlewares.CatchExceptionMiddleware',
    'tests.contrib.django.app.middlewares.MiddlewareBlast',
    'ddtrace.contrib.django.TraceExceptionMiddleware',
]
```
What happens is that our `TraceExceptionMiddleware` isn't called because the chain is short-circuited immediately. This results in a full trace where a 500 is reported but the `request_span.get_tag('error.stack')` is empty. Adding this extra middleware covers only the case where:
* there is an exception in a view
* a middleware returns a response before hitting the original `TraceMiddleware`.

We may think to handle that differently later on, like wrapping all registered middlewares here: https://github.com/django/django/blob/master/django/core/handlers/base.py#L25-L35 so that a second middleware is not required. Also, this would provide even more details about the time spent on Django / custom middlewares (this info is actually missing). Though it's more work and it could be a second step.

@LotharSee thoughts?